src/token_contract/src/main.nr

```diff
@@ -435,6 +340,47 @@ pub contract Token {
         finalization_payload.emit();
     }
 
+    #[contract_library_method]
+    fn subtract_balance(
+        context: &mut PrivateContext,
+        storage: Storage<&mut PrivateContext>,
+        account: AztecAddress,
+        amount: U128,
+        max_notes: u32,
+    ) -> U128 {
```

Comment:
this is the "super optimized" note fetching method, we're currently not using it on this version, and should be using it on every token transfer, if possible, i'd try to move it into `balance_set.nr` file

what this does is:
- start the recursion with 2 Notes
- if it's not enough, do recursion with 16 notes

ideally, this should be the normal behaviour of `<BalanceNote>.sub(...)` method, so we aren't "choosing when using it and when not" (we should use it always)